profile/src/main/java/rdx/works/profile/data/model/pernetwork/Network.kt

```diff
@@ -657,15 +705,9 @@ fun Profile.usedAccountDerivationIndices(
     }.map { it.derivationPathEntityIndex }.toSet()
 }
 
-fun Profile.nextAccountIndex(
-    derivationPathScheme: DerivationPathScheme,
-    forNetworkId: NetworkId? = null,
-    factorSourceID: FactorSource.FactorSourceID? = null
-): Int {
-    val network = networks.firstOrNull { it.networkID == forNetworkId?.value } ?: return 0
-    val factorSource = factorSources.find { it.id == factorSourceID } ?: mainBabylonFactorSource() ?: return 0
-    val accountsControlledByFactorSource = network.accounts.filter {
-        it.factorSourceId == factorSource.id && it.derivationPathScheme == derivationPathScheme
+fun Network.nextAccountIndex(factorSource: FactorSource): Int {
+    val accountsControlledByFactorSource = accounts.filter {
+        it.factorSourceId == factorSource.id && it.derivationPathScheme == DerivationPathScheme.CAP_26
```

Comment:
we need to find the index on the current network, so this is an extension for the `Network` and not for the `Profile`.
Same for the `nextAppearanceId`